src/components/App.vue

```diff
@@ -295,54 +295,44 @@ async function loadRemoteFilesFromURLParams(
   params: UrlParams,
   setError: (err: Error) => void
 ) {
-  const dataStore = useDatasetStore();
-
   const urls = wrapInArray(params.urls);
-  const names = wrapInArray(params.names ?? []);
-
-  const fileResults = await Promise.allSettled(
-    urls.map(async (url, index) => {
-      const { pathname } = new URL(url, window.location.href);
-      const name = names[index] || pathname.split('/').at(-1) || '';
-
-      return fetchFile(url, name);
-    })
-  );
-
-  const results = await Promise.allSettled(
-    fileResults.map(async (fileResult) => {
-      if (fileResult.status === 'fulfilled') {
-        const loadResults = await dataStore.loadFiles([fileResult.value]);
-        const errored = loadResults.find((s) => !s.loaded);
-        if (errored) {
-          // force the promise to error out
-          throw new Error();
-        }
-        // only select the first dataset
-        return loadResults[0];
-      }
-      throw fileResult.reason;
-    })
+  const names = wrapInArray(params.names);
```

Comment:
We still need `params.names ?? []` here, since if it's undefined then `names` becomes `[undefined]`.